### PR TITLE
Fix warnings for Elixir 1.4.0

### PR DESCRIPTION
**Pull request by xtian**
I fixed a few warnings that were being output for function/variable ambiguity and the `Dict` module.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  - 1.1.0
-  - 1.2.0
-otp_release:
-  - 17.4
-  - 18.1
+elixir: 1.4.0
+otp_release: 19.2
+matrix:
+  include:
+    - elixir: 1.1.0
```
Comment (doomspork) [inline .travis.yml]: We don't need to worry about 1.1 anymore but I think we should probably add coverage for 1.3 and 1.2, what do you think?

**Review by doomspork — CHANGES_REQUESTED** [review]
Thank you @xtian, one question with regards to Travis

File: .travis.yml (hunk 1 above)
Comment (xtian) [reply .travis.yml]: @doomspork Most of the projects under the elixir-lang org only test the oldest and newest supported versions. I'm not aware of anything which could fail for 1.3 but not 1.2 or 1.4. Do you know of anything?

**Comment by xtian** [comment]
@doomspork Any further thoughts on this? As I understand it, breakages relating to the Elixir version could be introduced in two ways:

- a new API/language feature could be used which would break for old versions which didn't support it
- some old API/language feature could be deprecated in a new version (such as the function/variable ambiguity fixed in this PR)

Both cases are covered by testing the oldest and newest supported versions. Anything else would be a semver violation by the Elixir project and should be fixed there.

**Comment by doomspork** [comment]
Thank you @xtian 👍

**Pull request merged**